### DISCUSSION TITLE
Script to clone print settings

### DISCUSSION THREAD

**Discussion by GVeno** [sn4k3/UVtools#345] [Scripts]
I wrote a little script that copies print settings from one file to others in the same directory or directory tree. Please let me know if you find it useful, or improve on it in any way. Thanks to Tiago for some much-needed help along the way. 

UPDATE 17-Nov-2021: I updated the script to not use parallel processing for the files. The latest is attached.

[ScriptCloneSettings.cs.txt](https://github.com/sn4k3/UVtools/files/7557510/ScriptCloneSettings.cs.txt)

**Comment by sn4k3** [comment]
Thanks for the contribution.
A few notes:

1. You could add a directory input box to the UI, i could add a directory picker input
2. Don't use Parallel to decode files, the decode method already parallel stuff and then you will exhaust the CPU making PC slow

Overall, nice work

**Comment by GVeno** [comment]
Thanks for your suggestions.


  1.  I’m OK with the location of the SlicerFile being the root of the tree of files to modify. You don’t need to add a directory picker for my sake.
  2.  I was wondering why the parallel for-each didn’t seem to speed things up at all. I’ll revert to a standard for-each.

-Gina

**Comment by GVeno** [comment]
I did the fix for 2. and replaced the upload in my original post.

**Reply by sn4k3** [reply]
Nice  👍 
I'm also adding new inputs to scripting, to add ease of use when they are required...

![UVtools_2021-11-17_20-32-24](https://user-images.githubusercontent.com/113281/142277891-9c0808a8-b210-4cc6-96bd-c7c1a424e37e.png)

**Reply by sn4k3** [reply]
New version is out, you can now open files partially. 
Use: `var file = FileFormat.Open(filePath, FileFormat.FileDecodeType.Partial, Progress);`
That helper returns null if no valid filepath
